consensus/src/alerts/mod.rs

```diff
@@ -490,3 +490,367 @@ pub(crate) async fn run<H: Hasher, D: Data, MK: MultiKeychain>(
+        let mut own_node = Alerter::new(
+            &key_boxes[own_index.0],
+            AlertConfig {
+                n_members,
+                session_id: 0,
+            },
+        );
+        let fork_proof =
+            make_fork_proof(forker_index, &key_boxes[forker_index.0], 0, n_members).await;
+        let empty_alert = Alert::new(double_committer, fork_proof.clone(), vec![]);
+        let empty_alert_hash = Signable::hash(&empty_alert);
+        let signed_empty_alert = Signed::sign(empty_alert.clone(), &key_boxes[double_committer.0])
+            .await
+            .into_unchecked();
+        assert_eq!(
+            own_node
+                .on_message(AlertMessage::ForkAlert(signed_empty_alert))
+                .await,
+            Some(AlerterResponse::ForkResponse(
+                Some(ForkingNotification::Forker(fork_proof.clone())),
+                empty_alert_hash,
+            )),
+        );
+        let forker_unit = fork_proof.0.clone();
+        let nonempty_alert = Alert::new(
+            double_committer,
+            fork_proof.clone(),
+            vec![forker_unit.clone()],
+        );
+        let nonempty_alert_hash = Signable::hash(&nonempty_alert);
+        let signed_nonempty_alert =
+            Signed::sign(nonempty_alert.clone(), &key_boxes[double_committer.0])
+                .await
+                .into_unchecked();
+        let signed_nonempty_alert_hash =
+            Signed::sign_with_index(nonempty_alert_hash, &key_boxes[double_committer.0])
+                .await
+                .into_unchecked();
+        let mut multisigned_nonempty_alert_hash = signed_nonempty_alert_hash
+            .check(&key_boxes[double_committer.0])
+            .expect("the signature is correct")
+            .into_partially_multisigned(&key_boxes[double_committer.0]);
+        for i in 1..n_members.0 - 2 {
+            let node_id = NodeIndex(i);
+            let signed_nonempty_alert_hash =
+                Signed::sign_with_index(nonempty_alert_hash, &key_boxes[node_id.0])
+                    .await
+                    .into_unchecked();
+            multisigned_nonempty_alert_hash = multisigned_nonempty_alert_hash.add_signature(
+                signed_nonempty_alert_hash
+                    .check(&key_boxes[double_committer.0])
+                    .expect("the signature is correct"),
+                &key_boxes[double_committer.0],
+            );
+        }
+        assert_eq!(
+            own_node
+                .on_message(AlertMessage::ForkAlert(signed_nonempty_alert))
+                .await,
+            None,
+        );
+    }
+
+    #[tokio::test]
+    async fn who_is_forking_ok() {
+        let n_members = NodeCount(7);
+        let own_index = NodeIndex(0);
+        let forker_index = NodeIndex(6);
+        let own_key_box = KeyBox::new(n_members, own_index);
+        let forker_key_box = KeyBox::new(n_members, forker_index);
+        let own_node = Alerter::new(
```

Comment:
Nit: we typically use the name "node" for a larger thing consisting of multiple components: `creator`, `terminal`, `alerter` and few others. It would be better to call it `own_alerter` or `alerter`.

Reply:
Yeah, that makes sense. I renamed it from `alerter` because some tests have a different mocked-up node that sends an alert and is called…alerter. So the thing that the alert comes from is not always the thing that I called “own_node”. I was confused by that for a while.

What about `this`? It kind of fits since the alerter is what's being tested

Reply:
Yes, it's fine. Feel free to resolve this comment by yourself, I don't have strong opinions about names, just wanted to point that there is this specific meaning of "node" :)

Reply:
Fixed in 27fcea1e! 😀